assets/src/edit-story/components/panels/link.js

```diff
@@ -82,29 +85,48 @@ function LinkPanel({ selectedElements, onSetProperties }) {
   );
   const handleSubmit = useCallback(
     (evt) => {
-      onSetProperties({ link: state?.url ? state : null });
+      onSetProperties({ link: state?.url ? { ...state } : null });
       if (evt) {
         evt.preventDefault();
       }
     },
     [state, onSetProperties]
   );
 
+  const {
+    actions: { getLinkMetadata },
+  } = useAPI();
+
   const canLink = selectedElements.length === 1 && !isFill;
-  const populateMetadata = useCallback(
-    debounce(300, async (/** url */) => {
-      // TODO(wassgha): Implement getting the page metadata
-    })
-  );
+
+  const [populateMetadata] = useDebouncedCallback((url) => {
+    if (!isValidUrl(withProtocol(url))) {
+      return;
+    }
+    setFetchingMetadata(true);
+    getLinkMetadata(withProtocol(url))
+      .then(({ title, image }) => {
+        setState((originalState) => ({
+          ...originalState,
+          desc: title ? title : originalState.desc,
+          icon: image
+            ? toAbsoluteUrl(withProtocol(originalState.url), image)
```

Comment:
Not sure what this is meant to do, but visibly it looks odd: first it adds the protocol and then tries to make an absolute URL. But a URL with a protocol is always absolute. No?

Reply:
It tries to make `image` an absolute URL, not `originalState.url`. `image` can sometimes be `/images/something.jpg` which btw I think should be solved on the API side, but till then this should work

Reply:
I agree: it should definitely be an API. But at this point I got your point. We can move with this and come back to it on API level.